### DISCUSSION TITLE
How to customize select field

### DISCUSSION THREAD

**Discussion by prithvi5481** [react-querybuilder/react-querybuilder#773] [Q&A]
can someone please explain how to customize select field using custom CSS or is there any other better way to customize it.

`import { useState } from 'react';
import type { Field, RuleGroupType } from 'react-querybuilder';
import { QueryBuilder, formatQuery } from 'react-querybuilder';
import 'react-querybuilder/dist/query-builder.css';
import '../../../../../app/globals.css';

const fields: Field[] = [
  { name: 'firstName', label: 'First Name' },
  { name: 'lastName', label: 'Last Name' },
];

const initialQuery: RuleGroupType = {
  combinator: 'and',
  rules: [
    { field: 'firstName', operator: 'beginsWith', value: 'Stev' },
    { field: 'lastName', operator: 'in', value: 'Vai,Vaughan' }
  ],
};

const QueryBuilderComponent = () => {
  const [query, setQuery] = useState(initialQuery);

  return (
    <div className='bg-background-offsetWeak p-4 rounded-lg'>
      <QueryBuilder
        fields={fields}
        query={query}
        onQueryChange={setQuery}
        controlClassnames={{
          queryBuilder: 'queryBuilder-branches',
        }}
      />
      <h4>Query</h4>
      <pre>
        <code className='text-text-primary'>{formatQuery(query, 'json')}</code>
      </pre>
    </div>
  );
};

export default QueryBuilderComponent;`

**Comment by jakeboone02** [comment]
Sorry it's taken so long to reply! Do these docs pages help?

- [Styling overview](https://react-querybuilder.js.org/docs/styling/overview)
- [CSS classes](https://react-querybuilder.js.org/docs/styling/classnames)
